docs: assumptions (#62)
freedomofpress/securedrop-protocol: README.md

```

## Assumptions

* **This is a cryptographic protocol conducted via a REST API.**  In theory, it
  is agnostic to the underlying transport.  In practice, all parties
  communicate with the server via HTTP over Tor.

> * **Message expiry/deletion will occur on a fuzzy interval.** The protocol will expire

Author: I think this would let us delete the "Message Retention" section under "Limitations and Discussion".

Reviewer: Good point - pulled in some of that info to this bullet point and deleted that section.

```

## Assumptions

* **This is a cryptographic protocol conducted via a REST API.**  In theory, it
  is agnostic to the underlying transport.  In practice, all parties

>   communicate with the server via HTTP over Tor.

Author: This prompts a useful rewrite.  Let me know what you think of d89593a63a9fbd5d65cec15dbbf90456d3cbb2b2.

* **Message expiry/deletion will occur on a fuzzy interval.** 
  The computation and bandwidth required for the message-fetching portion of this protocol limits the number of messages that can be stored on the server at once (a current estimate is that more than a few thousand would produce unreasonably slow computation times).
  The protocol will expire messages on the server at a fuzzy interval ``d`` days +/- ``i`` (for example, 37 +- 7 days would guarantee message availability for a minimum of 30 days). The goal of fuzzy-interval message expiry is to avoid writing precise metadata to disk about when a message was submitted, which would be implied by a fixed expiry time.

>   Client-side (local) message deletion will be supported for journalists. Note this is not an anti-forensic measure, because some indicator (such as a uuid hash) will be retained in order to prevent re-downloading a locally-deleted message.

Author: Here's a slight clarification for your consideration, @rocodes.  Intuitively, since UUIDs are easily enumerable, I'm not sure how meaningful it is to hash them, especially locally.  So I think the salient point is just that we imagine we'll have to keep *some* kind of tracking reference a message that's been deleted locally.

```suggestion
  Client-side (local) message deletion will be supported for journalists. Note this is not an anti-forensic measure, because some identifier for a locally-deleted message may be retained in order to avoid re-downloading it.
```

Reviewer: ah thanks, I think that was missing a slash (uuid or hash) - but I prefer your more general wording. Will include all your changes

Reviewer: see [a510f02](https://github.com/freedomofpress/securedrop-protocol/commit/a510f029fc8f11ff433ab0ea8450e58f7ac31766)

> * **Messaging an arbitrary subset of journalists will not be supported in the UI.**

Author: I think your inline phrasing re: lack of UI support is clearer.

```suggestion
* **Messaging an arbitrary subset of journalists will not be supported.**
```
